File: biddo_client/lib/core/models/exchange_rate.dart
import 'currency.dart';

class ExchangeRate {
  String id;
  String baseCurrencyId;
  Map<String, double> rates;

  ExchangeRate({
    required this.id,
    required this.baseCurrencyId,
    required this.rates,
  });

  static ExchangeRate fromJSON(dynamic data, List<Currency> currencies) {
    return ExchangeRate(
      id: data['id'],
      baseCurrencyId: data['baseCurrencyId'],
      rates: Map.fromEntries(
        currencies.map((currency) => MapEntry(
              currency.code,
              double.parse((data['rates'][currency.code] ?? 1.0).toString()),
            )),
      ),
    );
  }
}

File: core/domain/exchange_rate.ts
export class ExchangeRate {
  id: string
  baseCurrencyId: string
  rates: Record<string, number>

  constructor(params: { id: string; baseCurrencyId: string; rates: Record<string, number> }) {
    this.id = params.id
    this.baseCurrencyId = params.baseCurrencyId
    this.rates = params.rates
  }

  static fromJSON(data: Record<string, unknown> = {}): ExchangeRate {
    return new ExchangeRate({
      id: data.id as string,
      baseCurrencyId: data.baseCurrencyId as string,
      rates: data.rates as Record<string, number>,
    })
  }
}
